File: problem/Count_Submatrices_Less_k/countSubmatrices.c
int countSubmatrices(int **grid, int gridSize, int *gridColSize, int k)
{
    int ans = 0;
    int m = gridSize;
    int n = gridColSize[0];

    int **px = (int **)malloc((m + 1) * sizeof(int *));
    for (int i = 0; i <= m; i++)
    {
        px[i] = (int *)calloc((n + 1), sizeof(int));
    }

    for (int i = 0; i < m; i++)
    {
        for (int j = 0; j < n; j++)
        {
            px[i + 1][j + 1] = grid[i][j] + px[i][j + 1] + px[i + 1][j] - px[i][j];
            if (px[i + 1][j + 1] <= k)
            {
                ans++;
            }
        }
    }

    for (int i = 0; i <= m; i++)
    {
        free(px[i]);
    }
    free(px);

    return ans;
}

File: problem/Count_Submatrices_Less_k/countSubmatrices.js
/**
 * @param {number[][]} grid
 * @param {number} k
 * @return {number}
 */
var countSubmatrices = function(grid, k) {
    const m = grid.length, n = grid[0].length;
    let ans = 0;
    const px = Array.from({length: m+1}, () => new Array(n+1).fill(0));
    for (let i = 0; i < m; i++) {
        for (let j = 0; j < n; j++) {
            px[i+1][j+1] = grid[i][j] + px[i][j+1] + px[i+1][j] - px[i][j];
            if (px[i+1][j+1] <= k) ans++;
        }
    }
    return ans;
};